go/main.go

```diff
@@ -183,7 +195,101 @@ func buildQuery(tweet *twitterTweet) string {
 	return finalQuery
 }
 
-func runInserter(alphas []api.DgraphClient, c *y.Closer, tweets <-chan interface{}) {
+type writer struct {
+	dir   string
+	f     *os.File
+	w     *gzip.Writer
+	fsz   uint64
+	maxSz uint64
+}
+
+func newWriter() *writer {
+	var w writer
+	w.maxSz = uint64(opts.fileSize)
+	w.dir = opts.outFilesPath
+	w.newFiles()
+	return &w
+}
+
+func (w *writer) Write(buf []byte) (int, error) {
+	if w.f == nil {
+		w.newFiles()
+	}
+
+	sz, err := w.w.Write(buf)
+	if err != nil {
+		return sz, err
+	}
+	if atomic.AddUint64(&w.fsz, uint64(sz)) > w.maxSz {
+		w.Finish()
+		w.newFiles()
+	}
+	return sz, nil
+}
+
+func (w *writer) newFiles() {
+	id := atomic.AddUint64(&fid, 1)
+	fname := fmt.Sprintf("%06d", id)
+
+	absDir, err := filepath.Abs(opts.outFilesPath)
+	checkFatal(err, "absdir %s", absDir)
+	path := path.Join(absDir, fname+gzFileSuffix)
+
+	f, err := os.OpenFile(path, os.O_RDWR|os.O_CREATE|os.O_EXCL, 0666)
+	checkFatal(err, "Failed to open file %s", fname+gzFileSuffix)
+
+	gw := gzip.NewWriter(f)
+	gw.Header.Name = fname + jsonFileSuffix
+
+	w.f, w.w, w.fsz = f, gw, 0
+	atomic.AddUint32(&stats.NumFiles, 1)
+}
+
+func (w *writer) Finish() {
+	if w.f == nil && w.w == nil {
+		return
+	}
+	checkFatal(w.w.Flush(), "writer flush failed")
+	checkFatal(w.w.Close(), "writer close failed")
+	checkFatal(w.f.Close(), "file close failed")
+}
+func runWriter(c *z.Closer, tweets <-chan interface{}) {
+	defer c.Done()
+	w := newWriter()
+	defer func() {
+		w.Finish()
+	}()
+	for {
+		select {
+		case <-c.HasBeenClosed():
+			return
+		case jsn, more := <-tweets:
+			if !more {
+				return
+			}
+			var tweet anaconda.Tweet
+			switch msg := jsn.(type) {
+			case anaconda.Tweet:
+				tweet = msg
+			default:
+				// Not a tweet
+				atomic.AddUint32(&stats.ErrorsJSON, 1)
+			}
+
+			atomic.AddUint32(&stats.Tweets, 1)
+
+			data, err := json.Marshal(tweet)
```

Comment:
Consider: adding `data = append(data, '\n')` here would make it compatible with the [JSONLines](https://jsonlines.org/) format.